harvesters/src/main/java/org/fao/geonet/kernel/harvest/HarvestManagerImpl.java

```diff
@@ -627,16 +631,20 @@ public synchronized OperResult clearBatch(String id) throws Exception {
         elapsedTime = (System.currentTimeMillis() - elapsedTime) / 1000;
 
         // clear last run info
-        removeInfo(id, context.getUserSession().getUserId());
-        ah.emptyResult();
+        if(harvesterContext.getUserSession() != null ) {
+            // This is unexpected has harvesterContext is shared between threads that may
+            // by running in parallel
+            removeInfo(id, harvesterContext.getUserSession().getUserId());
+            ah.emptyResult();
+        }
```

Comment:
@josegar74 I am interest in your feedback here as the workflow looks suspect to me.

Reply:
Like maybe it is okay, but I think it should be using the current session context (from the user clicking the button). Not the shared harvest service context.

This is the `clearBatch(String id)` method.

Reply:
I am going to change this to use ``getServiceContext()`` as I am sure we just want to record the user that is clicking the button (so we know who removed the harvest results)....

Reply:
Yes, that method should record the user removing the records from the harvester.